integration/benchmark_test.go

```diff
@@ -0,0 +1,239 @@
+// +build performance
+
+package integration
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	_ "github.com/aquasecurity/fanal/analyzer/command/apk"
+	_ "github.com/aquasecurity/fanal/analyzer/library/bundler"
+	_ "github.com/aquasecurity/fanal/analyzer/library/cargo"
+	_ "github.com/aquasecurity/fanal/analyzer/library/composer"
+	_ "github.com/aquasecurity/fanal/analyzer/library/npm"
+	_ "github.com/aquasecurity/fanal/analyzer/library/pipenv"
+	_ "github.com/aquasecurity/fanal/analyzer/library/poetry"
+	_ "github.com/aquasecurity/fanal/analyzer/library/yarn"
+	_ "github.com/aquasecurity/fanal/analyzer/os/alpine"
+	_ "github.com/aquasecurity/fanal/analyzer/os/amazonlinux"
+	_ "github.com/aquasecurity/fanal/analyzer/os/debianbase"
+	_ "github.com/aquasecurity/fanal/analyzer/os/photon"
+	_ "github.com/aquasecurity/fanal/analyzer/os/redhatbase"
+	_ "github.com/aquasecurity/fanal/analyzer/os/suse"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/apk"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/dpkg"
+	_ "github.com/aquasecurity/fanal/analyzer/pkg/rpm"
+
+	"github.com/aquasecurity/fanal/analyzer"
+	"github.com/aquasecurity/fanal/cache"
+	"github.com/aquasecurity/fanal/extractor/docker"
+	"github.com/aquasecurity/fanal/types"
+	godeptypes "github.com/aquasecurity/go-dep-parser/pkg/types"
+	dtypes "github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type testCase struct {
+	name                 string
+	imageName            string
+	imageFile            string
+	expectedFiles        []string
+	expectedOS           analyzer.OS
+	expectedPkgsFromCmds string
+	expectedLibraries    string
+}
+
+func runChecksBench(j int, b *testing.B, ac analyzer.Config, ctx context.Context, tc testCase, d string, c cache.Cache) {
+	for i := 0; i < b.N; i++ {
+		var wg sync.WaitGroup
+		for i := 0; i <= j; i++ {
+			wg.Add(1)
+			go func(i int) {
+				defer wg.Done()
+
+				actualFiles, err := ac.Analyze(ctx, tc.imageFile)
+				require.NoError(b, err)
+				for file, _ := range actualFiles {
+					assert.Contains(b, tc.expectedFiles, file, tc.name)
+				}
+				assert.Equal(b, len(tc.expectedFiles), len(actualFiles), tc.name)
+
+				// check OS
+				osFound, err := analyzer.GetOS(actualFiles)
+				require.NoError(b, err)
+				assert.Equal(b, tc.expectedOS, osFound, tc.name)
+
+				// check Packages
+				actualPkgs, err := analyzer.GetPackages(actualFiles)
+				require.NoError(b, err)
+				data, _ := ioutil.ReadFile(fmt.Sprintf("testdata/goldens/%s.expectedpackages.golden", strings.ReplaceAll(tc.imageName, "/", "")))
+				var expectedPkgs []analyzer.Package
+				json.Unmarshal(data, &expectedPkgs)
```

Comment:
ditto: the cost of `ioutil.ReadFile` and `json.Unmarshal` are not related to a performance of `fanal`.